### PR TITLE
Refresh static contact form for /kernel/real-time

### PR DESCRIPTION
**Pull request by britneywwc**
## Done

- Update design for static form on `/kernel/real-time/contact-us`
- Design: https://www.figma.com/design/mxftgCWDJ6s3LrlaedREkP/24.10-Static-forms?node-id=1-3&t=wJYqljxJE6kXFa8h-0
- Make form labels more accessible
- Fix missing payload on marketo submission

## QA
- Go to [demo](https://ubuntu-com-14044.demos.haus/kernel/real-time/contact-us)
- Check that the demo adheres to the design
- Check that required fields work as expected
- Once submitted, check that payload is correct (included screenshots below)
- Check that form payload is submitted to marketo successfully

## Issue / Card

Fixes [WD-13103](https://warthogs.atlassian.net/browse/WD-13103)

## Screenshots
Screenshot 2024-07-09 at 3 33 55 PM
Screenshot 2024-07-09 at 3 34 03 PM

## Help

[QA steps](https://discourse.canonical.com/t/qa-steps/152) - [Commit guidelines](https://discourse.canonical.com/t/commit-guidelines/148)

[WD-12998]: https://warthogs.atlassian.net/browse/WD-12998?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[WD-13103]: https://warthogs.atlassian.net/browse/WD-13103?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by britneywwc** [comment]
@petesfrench @juanruitina This is ready for review

**Comment by akbarkz** [comment]
@britneywwc SCSS and Prettier CI actions are failing, can you check please?

**Comment by juanruitina** [comment]
It seems `fieldset` only picks up `legend` as the label if it's immediately nested under it, so we will need to be a bit more explicit. `aria-labelledby` should do the trick (code is just a suggestion, feel free to tweak as you wish)

Current implementation (see accessibility inspector on the right):
Screenshot 2024-07-10 at 14 04 32

With `aria-labelledby` (inspector does pick label):
Screenshot 2024-07-10 at 14 06 14

Unrelated: I think you should us `p-section--hero` in the first section so the spacing is right.

**Comment by petesfrench** [comment]
@juanruitina This feels weird being able to select both, the versions you use and, that you don't use any. I also wonder if this could lead to funneling problems for the marketing team
image

**Comment by juanruitina** [comment]
I would be OK with disabling "I don't use Ubuntu today" and "I don't know" if any of the fields above them is checked. Similarly, disabling all other fields when either "I don't use Ubuntu today" and "I don't know" are checked.

File: static/sass/styles.scss [hunk 1]
```diff
@@ -1659,3 +1664,10 @@ ol.ordered-counter {
     margin-right: 0.25rem;
   }
 }
+
+legend {
```
Comment (petesfrench) [inline static/sass/styles.scss:1668]: Could this extend the class we have [on vanilla](https://github.com/canonical/vanilla-framework/blob/a0d8f348e9bcd8f9f523eb356f95ca09b149f41d/scss/_base_forms.scss#L101) or better yet, could we make a push to upstream this? cc: @bartaz

File: templates/kernel/real-time/contact-us.html [hunk 2]
```diff
@@ -19,71 +23,91 @@ <h1>Run real-time Ubuntu in production</h1>
...
+          <div class="row--50-50">
+            <div class="col">
+              <legend class="p-heading--2">In which industry do you operate?</legend>
+            </div>
+            <div class="col">
+              <label class="p-radio">
+                <input class="p-radio__input"
+                       type="radio"
+                       aria-labelledby="your-industry-is-industrial"
+                       name="your-industry-is"
```
Comment (petesfrench) [inline templates/kernel/real-time/contact-us.html:54]: If you include a `name` attribute on these they will get attached to the payload directly, which can cause the form submission to fail. Just make sure they are getting attached 'Comments_from_lead__c' textarea
image

File: templates/kernel/real-time/contact-us.html [hunk 3]
```diff
@@ -487,165 +300,447 @@ <h2 class="p-heading--3">How are you currently meeting your latency requirements
...
+      <div class="p-section">
+        <hr class="is-fixed-width" />
+        <fieldset class="p-fieldset-section" id="kind-of-device" aria-required="true">
+          <div class="row--50-50">
+            <div class="col">
+              <legend class="p-heading--2 is-required">What kind of device are you using?</legend>
+            </div>
+            <div class="col">
+              <label class="p-checkbox">
+                <input required
```
Comment (petesfrench) [inline templates/kernel/real-time/contact-us.html:316]:
```suggestion
                <input
```
This is making that individual checkbox required. But there nature, checkboxes cannot be made to be required. You would have to check it with some custom JS

File: static/js/src/contact-forms.js [hunk 4]
```diff
@@ -1,49 +1,84 @@
...
+    "how-do-you-consume-open-source"
+  );
+  // Do you have specific compliance or hardening requirements?
+  const hardeningRequirementsFieldset = document.getElementById(
+    "hardening-requirements"
+  );
+  // Who is responsible for tracking, testing and applying CVE patches in a timely manner?
+  const responsibleForTracking = document.getElementById(
+    "responsible-for-tracking"
+  );
```
Comment (petesfrench) [inline static/js/src/contact-forms.js:48]: This is likely why only form fields with ID's are being sent.

I don't have full context on this work, so please let me know if I am wrong. But I think we can automate this by looping through each field set and extracting the heading and input/textareas.

**Comment by petesfrench** [comment]
@britneywwc  I had a little play around and was able to implement the pre-existing script [static-forms.js](https://github.com/canonical/ubuntu.com/blob/9daccb4421904ac03b07952cd40fc9435afdf424/static/js/src/static-forms.js), this looks for the `.js-formfield` class name and dynamically adds the heading and user's input to the 'Comments_for_lead' fieldset. I think this is a better approach, as it means we wont have to hard code any forms.

You can see what I did here: https://github.com/canonical/ubuntu.com/pull/14061/files# I have left comments to indicate the changes I made. This is super rough, so if you aren't sure about anything please reach out. Hope this helps.

File: templates/kernel/real-time/contact-us.html (hunk 2 above)
Comment (petesfrench) [reply templates/kernel/real-time/contact-us.html:54]: This comment can be ignored if you implement the script I suggested below as it strips 'name' attributes before submission.

File: static/sass/styles.scss (hunk 1 above)
Comment (britneywwc) [reply static/sass/styles.scss:1668]: Just pushed a PR to upstream this to vanilla!

**Comment by juanruitina** [comment]
Right now it is possible to select both "I don't use Ubuntu" and "I don't know", I think the other option should also be disabled together with the releases above. Feel free to change to UX+1 once you fix this. Thank you!

File: static/js/src/static-forms.js [hunk 5]
```diff
@@ -185,3 +186,60 @@ const forms = document.querySelectorAll("form[action='/marketo/submit']");
...
 }
+
+/**
+ *
+ * @param {*} fieldset
+ * @param {*} checklistItem
+ *
+ * Disable & enable Ubuntu versions checklist based on user selection
+ */
+function ubuntuVersionsChecklist(fieldset, checklistItem) {
```
Comment (petesfrench) [inline static/js/src/static-forms.js:197]: Looks good! but I would make it even more generic, there might be a case where another form wants to use the same pattern

**Review by akbarkz — APPROVED** [review]

**Pull request merged**